MahApps.Metro/Controls/RangeSlider.cs

```diff
@@ -2025,17 +2025,15 @@ private Double CalculateNextTick(Direction dir, double chekingValue, double dist
             if (!IsMoveToPointEnabled)
             {
                 //Check if current value is exactly Tick value or it situated between Ticks
-                if (!IsDoubleCloseToInt((chekingValue - Minimum) / TickFrequency))
+                if (!IsDoubleCloseToInt((chekingValue + distance - Minimum) / TickFrequency))
                 {
-                    double x = (chekingValue - Minimum) / TickFrequency;
+                    double x = (chekingValue + distance - Minimum) / TickFrequency;
                     distance = TickFrequency * (int)x;
                     if (dir == Direction.Increase)
                     {
                         distance += TickFrequency;
                     }
                     distance = (distance - Math.Abs(chekingValue - Minimum));
-                    _currenValue = 0;
```

Comment:
I was not sure about the return statement, I removed it and the algorithm now falls through to the other case. Seems to work as indended. What do you say? OK?